newsrevenuehub/rev-engine
# DEV-3691: DonationPageNavbar -> DonationPageHeader, add header alt text

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Migrates DonationPageNavbar to TypeScript, renaming it DonationPageHeader (to match DonationPageFooter). Adds alt text to the header logo. Removes some cruft around the old component.

#### Why are we doing this? How does it help us?

Improves AX of contribution pages.

#### How should this be manually tested? Please include detailed step-by-step instructions.

There are a lot of possible fields here :(.

- Header background color (defined in style)
- Header background image
- Header logo
- Header link
- Header logo alt text (this can only be set in Django admin currently)

These different options _shouldn't_ conflict with each other but things I would pay specific attention to are:

- A page with no logo but a link defined
- A page with a logo and link but no alt text defined (this defines a fallback one so it is interactible by assistive tech)
- A page with nothing at all customized
- A page with only a background color
- A page with only an image

Things to test:

1. Edit the page. Confirm that the header in the page preview is correct and updates correctly when changes are made, especially when a new logo or background image are set.
2. Confirm that the page header on the published contribution page is correct.
3. Confirm that the page header on the default thank you page is correct.
4. Confirm that the page header on the contributor portal is correct throughout the portal.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No. Allowing the user to set the header alt text is out of scope for this story.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

Slight test coverage gap on the background image because this is set in CSS only.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3691

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

See epic https://news-revenue-hub.atlassian.net/browse/DEV-3056
